std/math.d

```diff
@@ -6694,15 +6694,23 @@ bool approxEqual(T, U, V)(T lhs, U rhs, V maxRelDiff, V maxAbsDiff = 1e-5)
             //static assert(is(T : real) && is(U : real));
             if (rhs == 0)
             {
-                return fabs(lhs) <= maxAbsDiff;
+                static if (is(IntegralTypeOf!U))
+                    return abs(lhs) <= maxAbsDiff;
+                else
+                    return fabs(lhs) <= maxAbsDiff;
             }
             static if (is(typeof(lhs.infinity)) && is(typeof(rhs.infinity)))
             {
                 if (lhs == lhs.infinity && rhs == rhs.infinity ||
                     lhs == -lhs.infinity && rhs == -rhs.infinity) return true;
             }
-            return fabs((lhs - rhs) / rhs) <= maxRelDiff
-                || maxAbsDiff != 0 && fabs(lhs - rhs) <= maxAbsDiff;
+            static if (is(IntegralTypeOf!(typeof(lhs - rhs))))
+
+                return abs((lhs - rhs) / rhs) <= maxRelDiff
```

Comment:
I agree with the substance in this comment.

I think we should make overload
```D
bool approxEqual(T, U)(T lhs, U rhs) if(isIntegral!T && isIntegral!U) { return lhs == rhs;}
```
And guard the real implementation from accepting both integers. Other uses of `approxEqual` with 2 integral values are programming errors, IMO. This overload is solely for convenience in generic programming.